### PR TITLE
New output mechanism

### PR DESCRIPTION
**Pull request by troydai**
Allow command returns a CommandOutput object which defines the output format.

Also, keep reorganizing the commands.

### CODE REVIEW

File: src/a01/cli/command_definition.py [hunk 1]
```diff
@@ -39,14 +41,19 @@ def execute(self, args: argparse.Namespace):
 
         return self.func(**kwargs)
 
+    def output(self, arg: argparse.Namespace):
+        result = self.execute(arg)
+        if isinstance(result, CommandOutput):
```
Comment (mcardosos) [inline src/a01/cli/command_definition.py:46]: >isinstance [](start = 11, length = 10)

IsInstance?

File: src/a01/models/task_collection.py [hunk 2]
```diff
@@ -1,77 +0,0 @@
...
-            yield task
-
-    def post(self) -> None:
-        try:
-            payload = [t.to_dict() for t in self.tasks]
-            session.post(f'{self.endpoint_uri()}/run/{self.run_name}/tasks', json=payload).raise_for_status()
-        except HTTPError:
-            self.logger.debug('HttpError', exc_info=True)
-            raise ValueError('Failed to create tasks in the task store.')
-
```
Comment (mcardosos) [inline src/a01/models/task_collection.py:33]: This went to the dispatcher IIRC?

File: src/a01/commands/get_run.py [hunk 3]
```diff
@@ -25,37 +22,36 @@
...
-                for failure in tasks.get_failed_tasks():
-                    output_in_table(zip_longest(failure.get_table_header(), failure.get_table_view()), tablefmt='plain')
-                    output_in_table(await get_log_content_async(failure.log_resource_uri, session),
-                                    tablefmt='plain',
-                                    foreground_color=colorama.Fore.CYAN)
-                output_in_table(tasks.get_summary(), tablefmt='plain')
+                for failure in tasks_output.get_failed_tasks():
+                    output.append(TaskBriefOutput(failure))
+                    output.append(TaskLogOutput(await get_log_content_async(failure.log_resource_uri, session)))
+                output.append(TasksSummary(tasks))
```
Comment (mcardosos) [inline src/a01/commands/get_run.py:41]: >output.append(TasksSummary(tasks)) [](start = 16, length = 34)

Do you still need this if it already exists in line 35?

File: src/a01/commands/get_run.py [hunk 4]
```diff
@@ -25,37 +22,36 @@
...
+
+            if raw:
+                run = Run.from_dict(await session.get_json(f'run/{run_id}'))
+                output.append(JsonOutput(run.to_dict()))
 
             if recording:
-                print()
-                print('Download recordings ...')
-                for task in tasks.tasks:
```
Comment (mcardosos) [inline src/a01/commands/get_run.py:50]: >tasks.tasks [](start = 28, length = 11)

:sunglasses:

File: src/a01/cli/command_definition.py (hunk 1 above)
Comment (troydai) [reply src/a01/cli/command_definition.py:46]: Yeah is `isinstance` is a built in method. ...

File: src/a01/models/task_collection.py (hunk 2 above)
Comment (troydai) [reply src/a01/models/task_collection.py:33]: They don't need. The `requests` is going to be replaced by `aiohttp`. And the logic of making HTTP requests will be decoupled from the model.

File: src/a01/commands/get_run.py (hunk 3 above)
Comment (troydai) [reply src/a01/commands/get_run.py:41]: Yes. It is intended to have the output duplicate. The reason is that the output could be long so this save users from scroll back to the top.

**Comment by troydai** [comment]
If no more comments, I will merge then.

**Pull request merged**